cache/navigation.go

```diff
@@ -0,0 +1,44 @@
+package cache
+
+import (
+	"context"
+	"time"
+
+	dpcache "github.com/ONSdigital/dp-cache"
+	"github.com/ONSdigital/dp-topic-api/models"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// NavigationCache is a wrapper to dpcache.Cache which has additional fields and methods specifically for caching navigation data
+type NavigationCache struct {
+	*dpcache.Cache
+}
+
+// NewNavigationCache create a navigation cache object to be used in the service which will update at every updateInterval
+// If updateInterval is nil, this means that the cache will only be updated once at the start of the service
+func NewNavigationCache(ctx context.Context, updateInterval *time.Duration) (*NavigationCache, error) {
+	config := dpcache.Config{
+		UpdateInterval: updateInterval,
+	}
+
+	cache, err := dpcache.NewCache(ctx, config)
+	if err != nil {
+		logData := log.Data{
+			"config": config,
+		}
+		log.Error(ctx, "failed to create cache from dpcache", err, logData)
+		return nil, err
+	}
+
+	navigationCache := &NavigationCache{cache}
+
+	return navigationCache, nil
+}
+
+// AddUpdateFunc adds an update function to the homepage cache
+func (nc *NavigationCache) AddUpdateFunc(key string, updateFunc func() *models.Navigation) {
+	nc.UpdateFuncs[key] = func() (interface{}, error) {
```

Comment:
(suggestion): i guess you could refactor this to take generics. 

Like 
```
type example interface {
  int64 | flat64
}
func Test[T example] () (T, error) {
 var test T
return test, nil
}
```
Like this it will be specific on what this accepts rather than a generic interface ?

Reply:
Thanks, for now I will leave as it is. Maybe we can come back to this in a different PR